kingpin/actors/aws/cloudformation.py

```diff
@@ -1230,3 +1238,24 @@ def _execute(self):
         # This main method triggers the creation, deletion or update of the
         # stack as necessary.
         yield self._ensure_stack()
+
+
+def parse_s3_url(url):
+    """ Gets bucket name, region, and key from url
+
+    returns bucket name, region, key
+    """
+    # virtual host style
+    # https://bucket-name.s3-Region.amazonaws.com/key-name
+    match = re.search(r'^https?://(.+)\.s3\.([^.]+)\.amazonaws\.com/(.+)', url)
```

Comment:
1. Make the regex pattern a global so we don't have to re-compile it.
2. Use non-greedy matching for bucket (avoids bad key names) and use named groups (easier to see what we use after we match) like: `^https?://(?P<bucket>.+?)\.s3\.(?P<region>[^.]+)\.amazonaws\.com/(?P<key>.+)$`